torch/autograd/gradcheck.py

```diff
@@ -62,30 +63,52 @@ def get_numerical_jacobian(fn, input, target=None, eps=1e-3):
     x_tensors = iter_tensors(target, True)
     j_tensors = iter_tensors(jacobian)
 
-    def compute_gradient(x, idx, is_mkldnn=False):
-
-        def fn_out():
-            if not is_mkldnn:
-                # x is a view into input and so this works
-                return fn(input).clone()
-            else:
-                # convert the dense tensor back to have mkldnn layout
-                return fn([x.to_mkldnn()])
-
-        orig = x[idx].item()
-        x[idx] = orig - eps
-        outa = fn_out()
-        x[idx] = orig + eps
-        outb = fn_out()
-        x[idx] = orig
-        r = (outb - outa) / (2 * eps)
-        return r.detach().reshape(-1)
+    def update_jacobians(x, idx, d, d_idx, is_mkldnn=False):
+
+        def compute_gradient(delta):
+            def fn_out():
+                if not is_mkldnn:
+                    # x is a view into input and so this works
+                    return fn(input).clone()
+                else:
+                    # convert the dense tensor back to have mkldnn layout
+                    return fn([x.to_mkldnn()])
+
+            orig = x[idx].item()
+            x[idx] = orig - delta
+            outa = fn_out()
+            x[idx] = orig + delta
+            outb = fn_out()
+            x[idx] = orig
+            r = (outb - outa) / (2 * eps)
```

Comment:
so this works as expected only if delta is `eps` or `j*eps`? Seems pretty fragile.

Reply:
for both cases. It's because for the case of complex functions with input x + yj (where both x and y are real) and output s, we are interested in calculating ds/dx and ds/dy. ds/dx and ds/dy are further used to calculate wirtinger and conjugare wirtinger derivatives.

Reply:
I understand, but nothing prevents the caller from using some other argument, not `eps` or `j*eps`, in which case the result would be unexpected.

Reply:
yes you are right. added some comments and an assert based on your suggestion https://github.com/pytorch/pytorch/pull/43208/files#diff-0687821017cb7b20f80a3a8ab5daca97R7